pandas/core/algorithms.py

```diff
@@ -40,6 +41,94 @@
 # top-level algos #
 # --------------- #
 
+def checked_add_with_arr(arr, b, arr_mask=None, b_mask=None):
```

Comment:
Can you put this function somewhere more below in this file? As it does not belong to the "top-level algos" (the title just above this)